File: app/src/main/java/app/revanced/bilibili/patches/main/MainActivityDelegate.java
package app.revanced.bilibili.patches.main;

import android.app.AlertDialog;
import android.content.Context;
import android.content.Intent;

import com.bilibili.app.preferences.BiliPreferencesActivity;

import app.revanced.bilibili.patches.drawer.DrawerPatch;
import app.revanced.bilibili.settings.Settings;
import app.revanced.bilibili.settings.fragments.BiliRoamingSettingsFragment;
import app.revanced.bilibili.utils.Utils;
import tv.danmaku.bili.MainActivityV2;

public class MainActivityDelegate {

    public static void onCreate(MainActivityV2 activity) {
        DrawerPatch.onMainActivityCreate(activity);
        Utils.runOnMainThread(1000L, () -> showHintIfNeeded(activity));
    }

    public static void onStart(MainActivityV2 activity) {
        DrawerPatch.onMainActivityStart(activity);
    }

    public static boolean onBackPressed(MainActivityV2 activity) {
        return DrawerPatch.onMainActivityBackPressed(activity);
    }

    static void showHintIfNeeded(Context context) {
        if (Settings.SHOW_HINT.getBoolean()) return;
        new AlertDialog.Builder(context)
                .setTitle(Utils.getString("biliroaming_usage_hint_title"))
                .setMessage(Utils.getString("biliroaming_usage_hint_message"))
                .setPositiveButton(Utils.getString("biliroaming_jump_settings"), (dialog, which) -> {
                    Settings.SHOW_HINT.saveValue(true);
                    Intent intent = new Intent(context, BiliPreferencesActivity.class);
                    intent.putExtra("extra:key:fragment", BiliRoamingSettingsFragment.class.getName());
                    intent.putExtra("extra:key:title", Utils.getString("biliroaming_settings_title"));
                    context.startActivity(intent);
                })
                .setNegativeButton(Utils.getString("biliroaming_get_it"), (dialog, which) -> {
                    Settings.SHOW_HINT.saveValue(true);
                })
                .show();
    }
}

File: integrations/app/src/main/java/app/revanced/bilibili/patches/main/MainActivityDelegate.kt
package app.revanced.bilibili.patches.main

import android.app.AlertDialog
import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.annotation.Keep
import app.revanced.bilibili.patches.drawer.DrawerPatch
import app.revanced.bilibili.settings.Settings
import app.revanced.bilibili.settings.fragments.BiliRoamingSettingsFragment
import app.revanced.bilibili.utils.Utils
import app.revanced.bilibili.utils.constraintSize
import com.bilibili.app.preferences.BiliPreferencesActivity
import tv.danmaku.bili.MainActivityV2

object MainActivityDelegate {
    @Keep
    @JvmStatic
    fun onCreate(activity: MainActivityV2) {
        DrawerPatch.onMainActivityCreate(activity)
        Utils.runOnMainThread(1000L) {
            showHintIfNeeded(activity)
        }
    }

    @Keep
    @JvmStatic
    fun onStart(activity: MainActivityV2) {
        DrawerPatch.onMainActivityStart(activity)
    }

    @Keep
    @JvmStatic
    fun onBackPressed(activity: MainActivityV2): Boolean {
        return DrawerPatch.onMainActivityBackPressed(activity)
    }

    @Keep
    @JvmStatic
    fun onSaveInstanceState(activity: MainActivityV2, outState: Bundle) {
    }

    private fun showHintIfNeeded(context: Context) {
        if (Settings.ShowHint() || Utils.isHd()) return
        AlertDialog.Builder(context)
            .setTitle(Utils.getString("biliroaming_usage_hint_title"))
            .setMessage(Utils.getString("biliroaming_usage_hint_message"))
            .setPositiveButton(Utils.getString("biliroaming_jump_settings")) { _, _ ->
                Settings.ShowHint.save(true)
                val intent = Intent(context, BiliPreferencesActivity::class.java)
                intent.putExtra("extra:key:fragment", BiliRoamingSettingsFragment::class.java.name)
                intent.putExtra("extra:key:title", Utils.getString("biliroaming_settings_title"))
                context.startActivity(intent)
            }
            .setNegativeButton(Utils.getString("biliroaming_get_it")) { _, _ ->
                Settings.ShowHint.save(true)
            }.create().constraintSize().show()
    }
}
